server/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -329,6 +329,7 @@ protected AnalyzedRelation visitQuerySpecification(QuerySpecification node, Stat
 
         ExpressionAnalysisContext expressionAnalysisContext = context.expressionAnalysisContext();
         expressionAnalysisContext.windows(node.getWindows());
+        expressionAnalysisContext.setErrorOnUnknownObjectKey(statementContext.sessionContext().errorOnUnknownObjectKey());
```

Comment:
Might be worth to require the session setting within the `ExpressionAnalysisContext` constructor and make it final. During the analysis of a statement it should not be possible for the value to change.